controllers/openstackmachine_controller.go

```diff
@@ -335,20 +335,23 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 		return ctrl.Result{}, nil
 	}
 
-	if openStackMachine.Spec.FloatingIP != "" {
-		err = r.reconcileFloatingIP(computeService, networkingService, instance, openStackMachine, openStackCluster)
-		if err != nil {
-			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("FloatingIP cannot be reconciled: %v", err))
-			return ctrl.Result{}, nil
-		}
-	}
-
 	if openStackCluster.Spec.ManagedAPIServerLoadBalancer {
 		err = r.reconcileLoadBalancerMember(logger, osProviderClient, clientOpts, instance, clusterName, machine, openStackMachine, openStackCluster)
 		if err != nil {
 			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("LoadBalancerMember cannot be reconciled: %v", err))
 			return ctrl.Result{}, nil
 		}
+	} else if util.IsControlPlaneMachine(machine) {
+		fp, err := networkingService.GetOrCreateFloatingIP(openStackCluster, openStackCluster.Spec.ControlPlaneEndpoint.Host)
+		if err != nil {
+			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be created: %v", err))
```

Comment:
nit: seems some error might be caused by 'get floating ip'? so saying can't be created is not enough?

Reply:
updated to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/573/files#diff-26fe0ac50b059e6812598aabecee0870R347